client-java/controller/src/main/java/org/evomaster/client/java/controller/SutHandler.java

```diff
@@ -79,6 +79,67 @@ default void setupForGeneratedTest(){}
      * Execute the given data insertions into the database (if any)
      *
      * @param insertions DTOs for each insertion to execute
+     * @param previous an array of insertion results which were executed before this execution
+     * @return insertion execution results
      */
     InsertionResultsDto execInsertionsIntoDatabase(List<InsertionDto> insertions, InsertionResultsDto... previous);
+
+
+    /**
+     * <p>
+     * return an instance of a client of an RPC service.
+     * </p>
+     *
+     * <p>
+     * This method must be blocking until the SUT is initialized.
+     * </p>
+     *
+     * <p>
+     * This method is only required when the problem is RPC for the moment,
+     * otherwise return null
+     * </p>
+     *
+     * TOCHECK with Andrea if it is a proper place where we put this method
+     * this method is required
+     *
+     * @param interfaceName a full name of an interface
+     * @return a client which could send requests to the interface
+     */
+    default Object getRPCClient(String interfaceName){return null;}
+
+    /**
+     * <p>
+     * execute an RPC endpoint with evomaster driver
+     * </p>
+     *
+     * TODO Man: this might be removed later
+     *           or we could keep it as alternative
+     *
+     * @param json contains info of an RPC endpoint
+     * @return value returned by this execution. it is nullable.
+     */
+    default Object executeRPCEndpoint(String json) throws Exception {return null;}
+
+    /**
+     * <p>
+     * execute an RPC endpoint with evomaster driver
+     * </p>
+     *
+     * TOCHECK with Andrea if it is a proper place where we put this method
+     * this method is required
+     */
+    default void extractRPCSchema(){}
```

Comment:
but this does not return anything, ie, `void`?

Reply:
it will be used in the generated tests if the invocation of RPC endpoints in tests is proceeded with `executeRPCAction` method in driver.

Reply:
in that case, shouldn't `executeRPCAction` chech if  `extractRPCSchema` has been called, and, if not, call it?

Reply:
we could do that. `executeRPCAction` is also used during the search. during the search, `extractRPCSchema` would be invoked by `getSutInfo`, also inside `extractRPCSchema`, there is a check if the schema is extracted. if we move `extractRPCSchema` (totally doable), such check would be invoked every time during the search. 

currently, we keep the test generation with driver because we need it to process e2e tests since the other solution (test generated on the driver) only supports Java not kotlin.
if further we support kotlin, `extractRPCSchema` could be removed from `SutHandler`.